app/services/current-user.js

```diff
@@ -9,7 +9,10 @@ export default Service.extend({
     if (this.get('session.isAuthenticated')) {
       let user = this.get('store').queryRecord('user', {me: true});
       this.set('user', user);
-      user.catch(() => {
+      return user.then((user) => {
+        this.set('user', user);
```

Comment:
what's this all about? you're setting the `user` as the value of the promise and then overriding that with the resolve value? won't ember do the fulfillment behind the scenes?